src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -327,7 +327,13 @@ def __call__(
         image = self.vae.decode(latents).sample
 
         image = (image / 2 + 0.5).clamp(0, 1)
-        image = image.cpu().permute(0, 2, 3, 1).numpy()
+        image = image.cpu().permute(0, 2, 3, 1)
```

Comment:
```suggestion
        image = image.cpu().permute(0, 2, 3, 1).float().numpy()
```

instead of the if-else should be good here IMO

Reply:
Sticking with `float` here as this is a one time operation and is done on CPU.